File: JavaTest/src/IOCDomain/ErrorHandler.java
package IOCDomain;

import com.google.inject.Inject;

public class ErrorHandler implements IErrorHandler {
    public static int InstanceCount;
    ILogger _logger;

    @Inject
    public ErrorHandler(ILogger logger) {
        _logger = logger;
        InstanceCount++;
    }

    @Override
    public ILogger getLogger() {
        return _logger;
    }
}

File: IOCPerfTest/IOCDomain/ErrorHandler.cs
namespace IOCPerfTest.IOCDomain
{
	public class ErrorHandler : IErrorHandler
	{
	    public static int InstanceCount;

	    readonly ILogger _logger;

		public ErrorHandler(ILogger logger)
		{
			_logger = logger;
		    InstanceCount++;
		}

		public ILogger Logger { get { return _logger; } }
	}
}
